cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/web/HomeController.java

```diff
@@ -35,6 +53,40 @@ public String home(Model model) {
         //model.addAttribute("journals", pubDao.getJournals());
         model.addAttribute("journalCount", pubDao.getJournalCount());
         //model.addAttribute("textCountriesCount", pubDao.getCountriesInTextCount());
+        
+		HashMap<String, List<String>> countriesMap = new HashMap<String, List<String>>();
+		ArrayList<String> countriesList = new ArrayList<String>();
+        //Is Atomic Int the best way to this?
+        AtomicInteger highCount = new AtomicInteger(0);
+        AtomicInteger lowCount = new AtomicInteger(0);
+        try (CloseableIterator<CountriesImpl> countries = mongoTemplate.stream(new Query().noCursorTimeout(), CountriesImpl.class)) {

```

Comment:
business logic should not be done in controller!

Reply:
Where should this be done? In order to create the gradient for the map I need the high and low values. Should this be done else where and stored some place and recalled here? I also need to serialize all the country objects in a specific format to work with the map, should that not be done here as well?

Reply:
There should be a country service that calculates what you need here. Does this need to be done only once? If so it should be initiated in a `@PostConstruct` method and then the controller just asks for the results. Does this make sense?